src/cloudwatch.ts

```diff
@@ -0,0 +1,148 @@
+        cwLogger.info(`Retrieving count of events for ${req.logGroupName} with the filter: ${req.filterPattern}`);
+        const client: CloudWatchLogs = AWSXRay.captureAWSClient(new CloudWatchLogs(this.logConfig));
+        const eventCount: number[] = [];
+        let events = await client.filterLogEvents(req).promise();
+        eventCount.push(events.events!.length);
+        while (events.nextToken) {
+            req.nextToken = events.nextToken;
+            events = await client.filterLogEvents(req).promise();
+            eventCount.push(events.events!.length);
+        }
+        return [req.logGroupName, eventCount.reduce((total, num) => total + num, 0)];
+    }
+
+    /**
+     * Get total timeouts today
+     * @returns {Array<[string,number]>} An array of tuples containing the name of the lambda and the number of timeouts today.
+     */
+    public async getTimeouts(): Promise<Array<[string, number]>> {
+        const groups: string[] = [
+            `/aws/lambda/activities-${this.branch}`,
+            `/aws/lambda/atf-report-gen-${this.branch}`,
+            `/aws/lambda/atf-report-gen-init-${this.branch}`,
+            `/aws/lambda/auth-${this.branch}`,
+            `/aws/lambda/cert-gen-${this.branch}`,
+            `/aws/lambda/cert-gen-init-${this.branch}`,
+            `/aws/lambda/cert-gov-notify-${this.branch}`,
+            `/aws/lambda/defects-${this.branch}`,
+            `/aws/lambda/dlq-handler-${this.branch}`,
+            `/aws/lambda/doc-gen-${this.branch}`,
+            `/aws/lambda/logs-${this.branch}`,
+            `/aws/lambda/metrics-collector-${this.branch}`,
+            `/aws/lambda/preparers-${this.branch}`,
+            `/aws/lambda/retro-gen-${this.branch}`,
+            `/aws/lambda/retro-gen-init-${this.branch}`,
+            `/aws/lambda/technical-records-${this.branch}`,
+            `/aws/lambda/test-number-${this.branch}`,
+            `/aws/lambda/test-results-${this.branch}`,
+            `/aws/lambda/test-stations-${this.branch}`,
+            `/aws/lambda/test-types-${this.branch}`
+        ];
+        const dayMillis: number = this.now.startOf("day").toMillis();
+        return await Promise.all(groups.map(async (logGroup) => {
+            const filterReq: FilterLogEventsRequest = {
+                logGroupName: logGroup,
+                startTime: dayMillis,
+                filterPattern: "Task timed out"
+            };
+            return this.getLogEventsCount(filterReq);
+        }));
+    }
+
+    /**
+     * Pushes the metrics to CloudWatch.
+     * @public
+     * @async
+     * @param visitsToday Number of visits since 00:00 UTC today.
+     * @param oldVisits Number of visits since now UTC - 10 hours.
+     * @returns {string} Combination of visit metrics for response.
+     */
+    public async sendMetrics(visitsToday: number, oldVisits: number): Promise<string> {
+        const client: CloudWatch = AWSXRay.captureAWSClient(new CloudWatch(this.cwConfig));
```

Comment:
coded as current re AWSXRay.captureAWSClient will be difficult to test without an abstraction

Reply:
Testing with `AWS_XRAY_CONTEXT_MISSING=LOG_ERROR` is reccomended